stdlib/builtins.pyi

```diff
@@ -243,7 +243,7 @@ _LiteralInteger = _PositiveInteger | _NegativeInteger | Literal[0]  # noqa: Y026
 
 class int:
     @overload
-    def __new__(cls, /) -> Literal[0]: ...
+    def __new__(cls: type[int], /) -> Literal[0]: ...
```

Comment:
this doesn't stop the overload from applying to subclasses, because `type[bool]` is a subtype of `type[int]`

Reply:
Yeah I don't think the current type system has a way to fix the issue I brought up.